examples/stm32/xively_demo_stm32f4xx_nucleo_eth/Src/provisioning.c

```diff
@@ -0,0 +1,172 @@
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "provisioning.h"
+
+#if USER_DATA_WIFI_DEVICE
+#include "wifi_interface.h"
+static int8_t provisioning_get_ap_credentials( user_data_t* udata );
+#endif /* USER_DATA_WIFI_DEVICE */
+
+static inline void provisioning_print_banner( void );
+static int8_t provisioning_get_xively_credentials( user_data_t* udata );
+
+int8_t provisioning_start( user_data_t* dst )
+{
+    provisioning_print_banner();
+
+#if USER_DATA_WIFI_DEVICE
+    /* Get WiFi AP Credentials from the user */
+    switch ( provisioning_get_ap_credentials( dst ) )
+    {
+        case 1:
+            printf( "\r\n>> Saving user data to flash" );
+            user_data_save_to_flash( dst );
+            break;
+        case 0:
+            break;
+        default:
+            printf( "\r\n\t[ERROR] Getting AP credentials from user" );
+            return -1;
+    }
+#endif /* USER_DATA_WIFI_DEVICE */
+
+    /* Get Xively Credentials from the user */
+    switch ( provisioning_get_xively_credentials( dst ) )
+    {
+        case 1:
+            printf( "\r\n>> Saving user data to flash" );
+            user_data_save_to_flash( dst );
+            break;
+        case 0:
+            break;
+        default:
+            printf( "\r\n\t[ERROR] Getting AP credentials from user" );
+            return -1;
+    }
+    return 0;
+}
+
+#if USER_DATA_WIFI_DEVICE
+/**
```

Comment:
Good point 👍  I'll keep it in mind for the implementation of the `Common` folder